polars/polars-time/src/windows/groupby.rs

```diff
@@ -462,38 +462,10 @@ pub(crate) fn groupby_values_iter<'a>(
     tz: Option<Tz>,
 ) -> Box<dyn TrustedLen<Item = PolarsResult<(IdxSize, IdxSize)>> + 'a> {
     let mut offset = period;
-    offset.negative = !period.negative;
-    if offset.duration_ns() > 0 {
-        // t is at the right endpoint of the window
-        let iter = groupby_values_iter_lookbehind(period, offset, time, closed_window, tu, tz, 0);
-        Box::new(iter)
-    } else if closed_window == ClosedWindow::Right || closed_window == ClosedWindow::None {
-        // only lookahead
-        let iter = groupby_values_iter_full_lookahead(
-            period,
-            offset,
-            time,
-            closed_window,
-            tu,
-            tz,
-            0,
-            None,
-        );
-        Box::new(iter)
-    } else {
-        // partial lookahead
-        let iter = groupby_values_iter_partial_lookahead(
-            period,
-            offset,
-            time,
-            closed_window,
-            tu,
-            tz,
-            0,
-            None,
-        );
-        Box::new(iter)
-    }
+    offset.negative = true;
+    // t is at the right endpoint of the window
+    let iter = groupby_values_iter_lookbehind(period, offset, time, closed_window, tu, tz, 0);
+    Box::new(iter)
```

Comment:
so satisfying 😋

Reply:
So all this didn't hit any test? Just to be sure, windows may still be lagging right?

Reply:
There are three branches here:

- if period > 0
- if period == 0 and closed is right or both
- if period == 0 and closed is left or none

if `period == 0` is disallowed, then only the first one is necessary, and the `if` statement can be removed and the code detended

`offset` and `period` are the same here, they just differ in sign, because the `rolling_*` functions don't have an `offset` argument. Windows where `offset != period` (like, that for point `2022-01-05` the window is `[2022-01-03, 2022-01-04)`) are only possible for `groupby_rolling` (and that's unaffected by this PR)

Reply:
Thanks for explanation. Sorry for the questions on my side. I am working on many things, so I try to get an understanding without getting fully in the code. Looks good!

Reply:
Thanks! And please always do ask for clarifications when things aren't clear, this'll also help anyone in the future who ends up here from a `git blame` trying to figure out why a certain change was made